specification/batch/resource-manager/Microsoft.Batch/stable/2023-05-01/BatchManagement.json

```diff
@@ -0,0 +1,6405 @@
+      "type": "object",
+      "properties": {
+        "username": {
+          "type": "string",
+          "x-ms-client-name": "userName",
+          "title": "The user name to log into the registry server."
+        },
+        "password": {
+          "type": "string",
+          "title": "The password to log into the registry server.",
+          "x-ms-secret": true
+        },
+        "registryServer": {
+          "type": "string",
+          "title": "The registry URL.",
+          "description": "If omitted, the default is \"docker.io\"."
+        },
+        "identityReference": {
+          "title": "The reference to the user assigned identity to use to access an Azure Container Registry instead of username and password.",
+          "$ref": "#/definitions/ComputeNodeIdentityReference"
+        }
+      },
+      "title": "A private container registry."
+    },
+    "DiskEncryptionConfiguration": {
+      "type": "object",
+      "properties": {
+        "targets": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "x-nullable": false,
+            "title": "The disks to encrypt on each compute node.",
+            "description": "If omitted, no disks on the compute nodes in the pool will be encrypted.",
+            "enum": [
+              "OsDisk",
+              "TemporaryDisk"
+            ],
+            "x-ms-enum": {
+              "name": "DiskEncryptionTarget",
+              "modelAsString": false,
+              "values": [
+                {
+                  "value": "OsDisk",
+                  "description": "The OS Disk on the compute node is encrypted.",
+                  "name": "OsDisk"
+                },
+                {
+                  "value": "TemporaryDisk",
+                  "description": "The temporary disk on the compute node is encrypted. On Linux this encryption applies to other partitions (such as those on mounted data disks) when encryption occurs at boot time.",
+                  "name": "TemporaryDisk"
+                }
+              ]
+            }
+          },
+          "title": "The list of disk targets Batch Service will encrypt on the compute node",
+          "description": "On Linux pool, only \"TemporaryDisk\" is supported; on Windows pool, \"OsDisk\" and \"TemporaryDisk\" must be specified."
+        }
+      },
+      "description": "The disk encryption configuration applied on compute nodes in the pool. Disk encryption configuration is not supported on Linux pool created with Virtual Machine Image or Shared Image Gallery Image."
+    },
+    "ContainerConfiguration": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "title": "The container technology to be used.",
+          "enum": [
+            "DockerCompatible",
+            "CriCompatible"
+          ],
+          "x-ms-enum": {
+            "name": "ContainerType",
+            "modelAsString": false,
```

Comment:
Just out of curious, why set modelAsString false here ? I think extensible enum would be more acceptable. feel like fixed enum may cause some problems. The JS breaking is because m4 used to treat single value enum as constant type. and if adding one more values in it, then it no longer a constant. Given the fact, that you do have plan to extend the enum values, I think set modelAsString as true would be more reasonable.

Reply:
Thanks Qiao, I have made the change as suggestion that make modelAsString value true.